### PR TITLE
Draft: support association implementation

### PR DESCRIPTION
**Pull request by PoojaParlapalli**

### CODE REVIEW

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/mapping/modelToModel.pure [hunk 1]
```diff
@@ -480,7 +480,8 @@ function meta::pure::mapping::modelToModel::graphFetch::executionPlan::updateClu
 function meta::pure::mapping::modelToModel::graphFetch::executionPlan::generateInMemoryChildGraphNodes(tree: RoutedGraphFetchTree[1], parentPath: String[1], orderedPaths: String[*], debug: DebugContext[1]): InMemoryGraphFetchExecutionNode[*]
 {
    $tree.subTrees->cast(@RoutedPropertyGraphFetchTree)->map({st |
-      if (!$st.property->isPrimitiveValueProperty() && (!$st.property->instanceOf(QualifiedProperty) || $st.property->hasGeneratedMilestoningPropertyStereotype()),
+      let associationSubTree =$st->map(w|$w.propertyMapping->exists(r|$r.owner->toOne()->instanceOf(AssociationImplementation)));
+      if (!$st.property->isPrimitiveValueProperty() && !$associationSubTree && (!$st.property->instanceOf(QualifiedProperty) || $st.property->hasGeneratedMilestoningPropertyStereotype()),
```
Comment (PoojaParlapalli) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/mapping/modelToModel.pure:484]: try to make these changes in internal

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure [hunk 2]
```diff
@@ -48,7 +48,7 @@ function meta::pure::executionPlan::engine::java::graphFetch::inMemory::prepareF
    let currentNodeInfo = $node->match([
       r : InMemoryRootGraphFetchExecutionNode[1]        | let childNodeInfo   = $context->nodeInfosForPath($path + '.n1')->toOne();
                                                           let returnType      = if($r.checked, | ^CheckedJavaType(checkedOf = $simpleType, source = $childNodeInfo.returnType), | $simpleType);
-                                                          ^NodeInfo(path = $path, returnType = $returnType, graphFetchTrees = $childNodeInfo.graphFetchTrees);,
+                                                          ^NodeInfo(path = $path, returnType = $returnType, graphFetchTrees = $childNodeInfo.graphFetchTrees,data = $node.graphFetchTree);,
```
Comment (hardikmaheshwari) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure:51]: Lets add a comment that this data field is deprecated and should not be used

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure [hunk 3]
```diff
@@ -135,12 +135,15 @@ function <<access.private>> meta::pure::executionPlan::engine::java::graphFetch:
             let graphImpl      = $node->graphImplClass($path, $context);
             let currentSet     = $graphFetchTree.sets->toOne()->cast(@InstanceSetImplementation);
             let input          = j_parameter(javaObject(), 'input');
+            let globalObject   = javaField(['public'], javaHashMap(javaObject(),javaObject()), 'sharedObject',javaHashMap()->j_new([]));
```
Comment (hardikmaheshwari) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure:138]: Lets add a comment that this field is deprecated and should not be used

Comment (PoojaParlapalli) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure:138]: thank you, sure

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure (hunk 2 above)
Comment (PoojaParlapalli) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure:51]: thank you sure

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure [hunk 4]
```diff
@@ -619,7 +624,10 @@ function <<access.private>> meta::pure::executionPlan::engine::java::graphFetch:
       )
    };
    
-   let otherSetsFilterGeneration = $extensions.java_graphFetch_inMemory_applyTypeFilter->map(f|$f->evaluate([list($set), list($src), list($withoutFilter), list($context), list($debug)])->cast(@Function<{Nil[1]->Code[1..*]}>));
+   let filterNodeInfo = $context.nodeInfos->filter(z|$z.path == $path);
+   let filterContext  = ^GenerationContext(typeInfos=$context.typeInfos, conventions=$context.conventions, nodeInfos=$filterNodeInfo);
```
Comment (hardikmaheshwari) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure:628]: lets copy from $context
```^$context(nodeInfos=$filterNodeInfo)```

Comment (PoojaParlapalli) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/executionPlan/javaPlatform/executionNode/graphFetch/inMemory/graphFetchInMemory.pure]: thank you, made this change

**Pull request closed without merging**